File: tvos/plugin/ios/ReactNative/PlayerBridge.m
//
//  PlayerBridge.m
//  LightApp
//
//  Created by Afanasiev, Anatolii on 21/01/2020.
//  Copyright © 2020 Facebook. All rights reserved.
//

@import React;
#import <AVFoundation/AVFoundation.h>
#import "React/RCTBridgeModule.h"
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(PlayerBridge, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(playableItem, NSDictionary);
RCT_EXPORT_VIEW_PROPERTY(pluginConfiguration, NSDictionary);
RCT_EXPORT_VIEW_PROPERTY(onVideoEnd, RCTBubblingEventBlock);

@end

File: tvos/plugin/ios/ReactNative/PlayerBridge.swift
//
//  PlayerBridge.swift
//  LightApp
//
//  Created by Afanasiev, Anatolii on 21/01/2020.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation
import React

@objc(PlayerBridge)
class PlayerBridge: RCTViewManager {
    
    static let playerModuleName = "PlayerBridge"
    
    override public static func moduleName() -> String? {
        return PlayerBridge.playerModuleName
    }
    
    override public class func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    override open var methodQueue: DispatchQueue {
        return bridge.uiManager.methodQueue
    }
    
    override public func view() -> UIView? {
        guard let eventDispatcher = bridge?.eventDispatcher() else {
            return nil
        }
        return PlayerView(eventDispatcher: eventDispatcher)
    }
}
